kythe/typescript/indexer.ts

```diff
@@ -568,39 +634,32 @@ class Visitor {
    * can be optionally specified to help disambiguate nodes with multiple
    * declarations. See the documentation of Context for more information.
    */
-  getSymbolName(sym: ts.Symbol, ns: TSNamespace, context?: Context): VName {
-    let vnames = this.symbolNames.get(sym);
-    let declarations = sym.declarations;
+  getSymbolName(
+      sym: ts.Symbol, ns: TSNamespace, context: Context = Context.Any): VName {
+    const cached = this.symbolNames.get(sym, ns, context);
+    if (cached) return cached;
 
-    // Symbols with multiple declarations are disambiguated by the context
-    // they are used in.
-    const contextApplies = context !== undefined && declarations.length > 1;
-
-    if (!contextApplies && vnames && vnames[ns]) return vnames[ns]!;
-    // TODO: update symbolNames table to account for context kind
-
-    if (!declarations || declarations.length < 1) {
+    let declarations = sym.declarations;
+    if (declarations.length < 1) {
       throw new Error('TODO: symbol has no declarations?');
     }
 
-    // Disambiguate symbols with multiple declarations using a context. This
-    // only applies to getters and setters currently.
-    if (contextApplies) {
+    // Disambiguate symbols with multiple declarations using a context.
+    if (sym.declarations.length > 1) {
       switch (context) {
+        // Getter and setter contexts are filtered for getter and setter
+        // declarations, respectively.
         case Context.Getter:
           declarations = declarations.filter(ts.isGetAccessor);
           break;
         case Context.Setter:
           declarations = declarations.filter(ts.isSetAccessor);
           break;
+        // Otherwise, make no disambiguation.
+        default:
+          break;
       }
     }
-    // Otherwise, if there are multiple declarations but no context is
```

Comment:
Will deleting this codepath change behavior? Does it matter?
Now that you're defaulting to Context.Any, won't it change the behavior if there are multiple declarations, from using the get accessor to using whichever decl happens to be first?

Reply:
No, it doesn't change the behavior. This is an artifact of the work done to emit an implicit property on getters and setters:

https://github.com/kythe/kythe/blob/f8c4e9bb682adb3b2041e49d72029e6be3977e5f/kythe/typescript/indexer.ts#L1168-L1174

that we never removed.